graylog2-server/src/main/java/org/graylog2/web/resources/AppConfigResource.java

```diff
@@ -59,29 +59,27 @@ public String get(@Context HttpHeaders headers) {
             throw new RuntimeException("Unable to read AppConfig template while generating web interface configuration: ", e);
         }
 
-        final String serverTransportUri = serverUriFromHeadersOrDefault(headers);
         final Map<String, Object> model = new HashMap<String, Object>() {{
             put("rootTimeZone", configuration.getRootTimeZone());
-            put("serverUri", serverTransportUri);
+            put("serverUri", buildEndpointUri(headers));
             put("appPathPrefix", "");
         }};
         return engine.transform(template, model);
     }
 
-    private String serverUriFromHeadersOrDefault(HttpHeaders httpHeaders) {
-        final List<String> overrideServerUriHeaders = httpHeaders.getRequestHeader(CK_OVERRIDE_SERVER_URI);
-        if (overrideServerUriHeaders != null && !overrideServerUriHeaders.isEmpty()) {
-            Optional<String> firstHeader = overrideServerUriHeaders.stream().filter(s -> {
+    private String buildEndpointUri(HttpHeaders httpHeaders) {
+        Optional<String> endpointUri = Optional.empty();
+        final List<String> headers = httpHeaders.getRequestHeader(CK_OVERRIDE_SERVER_URI);
+        if (headers != null && !headers.isEmpty()) {
+            endpointUri = headers.stream().filter(s -> {
                 try {
-                    return !Strings.isNullOrEmpty(s) && new URL(s) != null;
-                } catch (MalformedURLException e) {
+                    return new URI(s).isAbsolute();
+                } catch (NullPointerException | URISyntaxException e) {
```

Comment:
For the reasons you have mentioned, this is an unnecessary change. The URL class does not have `isAbsolute`, because for an URL the scheme is mandatory in the first place. This was checked before by catching the `MalformedURLException` exception. Also, the condition might be shorter, but I really dislike the implicit null check by catching the exception, it makes it much less explicit to read.

Also, if we declare in the config file that `web_endpoint_uri` is an URI, we also need to mention that the scheme is mandatory if we assume this.